<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <title>Title</title>
  </head>
  <body>

  </body>
</html>
<script>
  (function foo() {

    (function bar() {
      console.log('我是bar方法中的代码')

      // 未捕获异常会导致退出一个或多个执行上下文
      throw 'Exit from bar and foo contexts'

      // 当捕获了异常，则不会影响后面的执行上下文
      // try {
      //   throw 'Exit from bar and foo contexts'
      // } catch (err) {
      //   console.log('catch error:', err)
      // }
    })()

    console.log('方法代码')
  })()

  console.log('全局代码')
</script>